core/src/test/java/io/grpc/internal/SpiffeIdParserTest.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.internal;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+
+@RunWith(Enclosed.class)
+public class SpiffeIdParserTest {
+
+  @RunWith(Parameterized.class)
+  public static class ParseSuccessTest {
+    @Parameter
+    public String uri;
+
+    @Parameter(1)
+    public String trustDomain;
+
+    @Parameter(2)
+    public String path;
+
+    @Test
+    public void parseSuccessTest() {
+      SpiffeIdParser.SpiffeId spiffeId = SpiffeIdParser.parse(uri);
+      assertEquals(trustDomain, spiffeId.getTrustDomain());
+      assertEquals(path, spiffeId.getPath());
+    }
+
+    @Parameters(name = "spiffeId={0}")
+    public static Collection<String[]> data() {
```

Comment:
Since we're testing against spec's '[a-zA-Z0-9.-_]', I can imagine to make a typo in a regexp and start/end it with a wrong symbol, miss a hyphen, or forget a whole rule (like A-Z) In that case, tests like "spiffe://trustdomain/abz", "spiffe://trustdomain/ABZ", "spiffe://trustdomain/019", and "spiffe://trustdomain/.-_" make sense (or actually just        "spiffe://trustdomain/abzABZ019.-_" includes all the checks).
I'm not sure I understand what potential bug we're covering by including the whole sequences (I assume we also need the one with uppercase alphabet in case if we want this pattern)

Reply:
Aligned with existing tests based on offline discussion